src/test/unit/StrategyManagerUnit.t.sol

```diff
@@ -1,69 +1,35 @@
 // SPDX-License-Identifier: BUSL-1.1
 pragma solidity =0.8.12;
 
-import "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
-import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
 import "@openzeppelin/contracts/mocks/ERC1271WalletMock.sol";
-
-import "forge-std/Test.sol";
-
-import "../../contracts/core/StrategyManager.sol";
-import "../../contracts/strategies/StrategyBase.sol";
-import "../../contracts/permissions/PauserRegistry.sol";
-import "../mocks/DelegationManagerMock.sol";
-import "../mocks/SlasherMock.sol";
-import "../mocks/EigenPodManagerMock.sol";
-import "../mocks/Reenterer.sol";
-import "../mocks/Reverter.sol";
-
-import "../mocks/ERC20Mock.sol";
-
-import "./Utils.sol";
-
-contract StrategyManagerUnitTests is Test, Utils {
-    Vm cheats = Vm(HEVM_ADDRESS);
-
-    uint256 public REQUIRED_BALANCE_WEI = 32 ether;
-
-    ProxyAdmin public proxyAdmin;
-    PauserRegistry public pauserRegistry;
-
+import "src/contracts/core/StrategyManager.sol";
+import "src/contracts/strategies/StrategyBase.sol";
+import "src/contracts/permissions/PauserRegistry.sol";
+import "src/test/mocks/ERC20Mock.sol";
+import "src/test/mocks/Reverter.sol";
+import "src/test/mocks/Reenterer.sol";
+import "src/test/utils/EigenLayerUnitTestSetup.sol";
+
+/**
+ * @notice Unit testing of the StrategyMananger contract, entire withdrawal tests related to the
+ * DelegationManager are not tested here but callable functions by the DelegationManager are mocked and tested here.
+ * Contracts tested: StrategyManager.sol
+ * Contracts not mocked: StrategyBase, PauserRegistry
+ */
+contract StrategyManagerUnitTests is EigenLayerUnitTestSetup {
     StrategyManager public strategyManagerImplementation;
     StrategyManager public strategyManager;
-    DelegationManagerMock public delegationManagerMock;
-    SlasherMock public slasherMock;
-    EigenPodManagerMock public eigenPodManagerMock;
 
+    IERC20 public dummyToken;
     StrategyBase public dummyStrat;
     StrategyBase public dummyStrat2;
     StrategyBase public dummyStrat3;
 
-    IStrategy public beaconChainETHStrategy;
-
-    IERC20 public dummyToken;
-
     Reenterer public reenterer;
```

Comment:
Nit: to save deploy time, you could declare this and deploy in the test contracts that only test reentrancy instead of deploying it for every test contract

Reply:
I believe Reenterer is currently setup that way, just deployed per unit test as needed. Its not being deployed every `setUp()` call before each test